File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/tasks/TasksKey.java
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.tasks;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.View;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackdemoexamplefragments.R;
import com.zhuinden.simplestackdemoexamplefragments.application.BaseKey;

/**
 * Created by Zhuinden on 2017.01.25..
 */

@AutoValue
public abstract class TasksKey
        extends BaseKey {
    public static TasksKey create() {
        return new AutoValue_TasksKey(R.layout.path_tasks);
    }

    @Override
    protected Fragment createFragment() {
        return new TasksFragment();
    }

    @Override
    public int menu() {
        return R.menu.tasks_fragment_menu;
    }

    @Override
    public boolean isFabVisible() {
        return true;
    }

    @Override
    public int navigationViewId() {
        return R.id.list_navigation_menu_item;
    }

    @Override
    public boolean shouldShowUp() {
        return false;
    }

    @NonNull
    @Override
    public View.OnClickListener fabClickListener(@NonNull Fragment f) {
        return v -> {
            TasksFragment fragment = (TasksFragment) f;
            fragment.openAddNewTask();
        };
    }

    @Override
    public int fabDrawableIcon() {
        return R.drawable.ic_add;
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/tasks/TasksKey.kt
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.tasks

import android.support.v4.app.Fragment
import android.view.View
import com.zhuinden.simplestackdemoexamplefragments.R
import com.zhuinden.simplestackdemoexamplefragments.application.BaseKey
import kotlinx.android.parcel.Parcelize

/**
 * Created by Zhuinden on 2018. 08. 20.
 */

@Parcelize
data class TasksKey(val placeholder: String = "") : BaseKey() {
    constructor() : this("")

    override fun layout(): Int = R.layout.path_tasks

    override val isFabVisible: Boolean
        get() = true

    override fun createFragment(): Fragment = TasksFragment()

    override fun menu(): Int = R.menu.tasks_fragment_menu

    override fun navigationViewId(): Int = R.id.list_navigation_menu_item

    override fun shouldShowUp(): Boolean = false

    override fun fabClickListener(f: Fragment): View.OnClickListener =
        View.OnClickListener { v ->
            val fragment = f as TasksFragment
            fragment.openAddNewTask()
        }

    override fun fabDrawableIcon(): Int = R.drawable.ic_add
}
